### PR TITLE
Switching historical time periods

### PR DESCRIPTION
**Pull request by tmasternak**
Requires: https://github.com/Particular/ServiceControl.Monitoring/pull/43

## Description
This PR introduces `historical-period-chooser` directive that is shared between endpoint list and details pages. It enables selecting historical period. When period is chosen in changes `history` query parameter used in query url and forces immediate requery action.

### CODE REVIEW

**Comment by WilliamBZA** [comment]
I'll merge this if no objections are raised or feedback given by the end of today (UTC time).

File: src/ServicePulse.Host/app/js/directives/ui.particular.graph.js [hunk 1]
```diff
@@ -73,9 +73,14 @@
                             .attr("font-family", "sans-serif")
                             .text(average.toFixed(2));
 
+                        var scaleYValue = scaleY(average == 0 ? 1 : average) + 3;
+                        if (isNaN(scaleYValue)) {
```
Comment (WojcikMike) [inline src/ServicePulse.Host/app/js/directives/ui.particular.graph.js]: Do we want to replace it with `scaleYValue | 0`. It's cosmetic so I can live with isNaN

File: src/ServicePulse.Host/app/js/views/endpoint_details/endpoint_details.controller.js [hunk 2]
```diff
@@ -4,41 +4,73 @@
...
+        var subscription;
 
-        var subscription = monitoringService.endpointDetails($routeParams.endpointName, $routeParams.sourceIndex).subscribe(function (endpointInstances) {
-            if (endpointInstances.error) {
-                // show warning
-                toastService.showWarning('Could not load endpoint details');
-            } else {
-                $scope.endpointInstances = endpointInstances;
-                $scope.loading = false;
+        $scope.periods = [
```
Comment (WojcikMike) [inline src/ServicePulse.Host/app/js/views/endpoint_details/endpoint_details.controller.js]: How about we define it as contestant and reuse it in endpoint monitoring and endpoint details?

File: src/ServicePulse.Host/app/js/views/endpoint_details/endpoint_details.controller.js [hunk 3]
```diff
@@ -4,41 +4,73 @@
...
-                    // Warn user
+            subscription = monitoringService.createEndpointDetailsSource($routeParams.endpointName, $routeParams.sourceIndex, $scope.selectedPeriod.value).subscribe(function (endpointInstances) {
+                if (endpointInstances.error) {
+                    toastService.showWarning('Could not load endpoint details');
+                } else {
+                    $scope.endpointInstances = endpointInstances;
+                    $scope.loading = false;
+                }
+
+                $scope.endpointInstances.forEach(function (instance) {
```
Comment (WojcikMike) [inline src/ServicePulse.Host/app/js/views/endpoint_details/endpoint_details.controller.js]: BTW. I know that we are not to concerned with performance of SP-SC communication but don't we want change API to make it a 1 call with an array of instanceID's?

Comment (WilliamBZA) [inline src/ServicePulse.Host/app/js/views/endpoint_details/endpoint_details.controller.js]: I guess that makes sense. Happy for me to leave it as is for now though?

**Comment by WilliamBZA** [comment]
I've made 2 of your suggested changes @WojcikMike

**Pull request merged**